### PR TITLE
Issue 154 - Added transition animation when going from and to the place detail view

### PR DESCRIPTION
**Pull request by sleroux**
First pass at creating the custom presenting transition between the map/detail view. I tried to make it as close as I could to the mock up with some exceptions for new things like the map button.

### CODE REVIEW

File: Prox/Prox.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -984,7 +992,7 @@
 				CLANG_WARN__DUPLICATE_METHOD_MATCH = YES;
 				"CODE_SIGN_IDENTITY[sdk=iphoneos*]" = "iPhone Developer";
 				COPY_PHASE_STRIP = NO;
-				DEBUG_INFORMATION_FORMAT = dwarf;
+				DEBUG_INFORMATION_FORMAT = "dwarf-with-dsym";
```
Comment (sleroux) [inline Prox/Prox.xcodeproj/project.pbxproj]: Added this so I can see symbols when profiling.

File: Prox/Prox/Utilities/Animatable.swift [hunk 2]
```diff
@@ -0,0 +1,11 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Foundation
+
+/// A protocol for to adhere to for anything that we want to animate
+protocol Animatable {
```
Comment (sleroux) [inline Prox/Prox/Utilities/Animatable.swift]: An interface I made to return back animation property structs so I don't need to expose views/variables in the package space when I don't need to just for animations.

File: Prox/Prox/PlaceCarousel/PlaceCarouselViewController.swift [hunk 3]
```diff
@@ -95,6 +90,12 @@ class PlaceCarouselViewController: UIViewController {
         }
     }
 
+    // fake the location to Hilton Waikaloa Village, Kona, Hawaii
+    fileprivate var fakeLocation: CLLocation = CLLocation(latitude: 19.924043, longitude: -155.887652)
+
+    fileprivate var monitoredRegions: [String: GeofenceRegion] = [String: GeofenceRegion]()
+    fileprivate var timeAtLocationTimer: Timer?
```
Comment (fluffyemily) [inline Prox/Prox/PlaceCarousel/PlaceCarouselViewController.swift]: So, these should no longer be in `PlaceCarouselViewController` as all location monitoring has moved to `LocationMonitor` I'm guessing a rebase went awry

File: Prox/Prox/PlaceDetails/PlaceDetailsDescriptionView.swift [hunk 4]
```diff
@@ -130,6 +130,7 @@ class PlaceDetailsDescriptionView: UIView {
             expandableLabelHeightConstraint.isActive = true
             logoBottomConstraint.constant = 0
         }
-        layoutIfNeeded()
+
+        setNeedsLayout()
```
Comment (fluffyemily) [inline Prox/Prox/PlaceDetails/PlaceDetailsDescriptionView.swift]: it's needed only for when we expand and contract the views once displayed to apply the layout changes, not required during setup of the views.

**Comment by fluffyemily** [comment]
Animation link:  https://drive.google.com/a/mozilla.com/file/d/0B883AVKFv0O5X2VSY3FiaVRUc1U/view?usp=sharing

**Comment by sleroux** [comment]
I can't seem to reproduce the flashing on my iPhone 6s (10.1) simulator or device. Are you seeing this flash on device as well?

**Comment by fluffyemily** [comment]
It doesn't happen on my 6S device or the 6 simulator. It happens on the 6+, 7 & 7+ simulators (I don't have devices for those sizes)

**Comment by sleroux** [comment]
@fluffyemily Should be good for another pass.

**Comment by fluffyemily** [comment]
@sleroux, that is now just beautiful on every device. Thank you.

Just rebase and then merge please :)

**Pull request merged**